### PR TITLE
Drop v1 tests, use only one scarb version

### PR DESCRIPTION
**Pull request by THenry14**
Closes https://github.com/foundry-rs/starknet-foundry/issues/813

## Introduced changes

- drop v1 tests, simplify setup, do not require asdf for scarb version management

## Checklist

- [x] Linked relevant issue
- [x] Updated relevant documentation
- [x] Added relevant tests
- [x] Performed self-review of the code
- [x] Added changes to `CHANGELOG.md`

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -19,6 +19,13 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 - Spying events interface is updated to enable the use of events defined inside contracts in assertions
 
+### Cast
+
+#### Changed
+
+- dropped official support for cairo 1 contracts. while they still should be working without any problems, from now on 
+the only officially supported cairo version is 2
```
Comment (war-in) [inline CHANGELOG.md:27]: I would rephrase it because I'm not sure that we are now on Cairo 2 😅 It would be better to call it `cairo compiler version 2` or something like that

File: crates/cast/tests/e2e/account/add.rs [hunk 2]
```diff
@@ -59,10 +59,14 @@ pub async fn test_happy_case() {
...
-        CONTRACTS_DIR.to_string() + "/v1/map",
+        CONTRACTS_DIR.to_string() + "/map",
         "put",
-        "3",
+        "30",
     ));
+    //     let tmp_dir = TempDir::new()?;
+    //     tmp_dir
+    //         .copy_from(&tests_folder_path, &["**/*.cairo"])
+    //         .context("Unable to copy files to temporary directory")?;
```
Comment (war-in) [inline crates/cast/tests/e2e/account/add.rs:69]: 👀

File: crates/cast/tests/e2e/account/create.rs [hunk 3]
```diff
@@ -91,9 +91,9 @@ pub async fn test_happy_case_generate_salt() {
 #[tokio::test]
 pub async fn test_happy_case_add_profile() {
     let current_dir = Utf8PathBuf::from(duplicate_directory_with_salt(
-        CONTRACTS_DIR.to_string() + "/v1/balance",
+        CONTRACTS_DIR.to_string() + "/map",
         "put",
-        "1",
+        "10",
```
Comment (war-in) [inline crates/cast/tests/e2e/account/create.rs:96]: Hmm, why is it changed? 🤔

File: crates/cast/tests/utils/build_contracts.sh [hunk 4]
```diff
@@ -1,31 +1,19 @@
...
-        pushd "$contract_dir"
-        scarb build
-        popd
-      fi
-    done
-
-  else
-    echo "Please run tests/utils/prepare_for_tests.sh script first!"
+for contract_dir in "$CONTRACTS_DIRECTORY"/*; do
+  if ! test -d "$contract_dir"/target && [[ "$contract_dir" != *"fails"* ]]; then
```
Comment (war-in) [inline crates/cast/tests/utils/build_contracts.sh:14]: Now I wonder if we should exclude already built directories. If someone is running this script he usually want to build them

Comment (THenry14) [reply crates/cast/tests/utils/build_contracts.sh:14]: This script is being called by our test suite if I remember correctly, not sure compiling these contracts each time we want to run test is preferable (especially so that these contracts generally don't change)

File: crates/cast/tests/e2e/account/create.rs (hunk 3 above)
Comment (THenry14) [reply crates/cast/tests/e2e/account/create.rs:96]: two weeks passed, and I don't remember 😅 my guess is a few more tests used this account and something failed because of too few funds, but not sure. If you want, I can try anc check it, but I don't think this is that relevant tbh :+1:

File: crates/cast/tests/e2e/account/add.rs (hunk 2 above)
Comment (THenry14) [reply crates/cast/tests/e2e/account/add.rs]: sorry

**Review by war-in — APPROVED** [review]
